src/main/java/org/mondemand/Client.java

```diff
@@ -1267,7 +1278,7 @@ private synchronized void dispatchStatsSamples() {
 
       // snapshot samples map for dispatch
       SamplesMessage[] samplesMsgs = this.samples.values().toArray(new SamplesMessage[0]);
-      this.samples = new ConcurrentHashMap<String, SamplesMessage>();
+      this.samples = new ConcurrentHashMap<String, SamplesMessage>(this.samples.size());
```

Comment:
I figured the previous size is a good initial size for the new map to avoid resizing.